Source/JavaScriptCore/dfg/DFGPredictionPropagationPhase.cpp

```diff
@@ -594,26 +594,44 @@ class PredictionPropagationPhase : public Phase {
                 break;
             case Array::Float32Array:
             case Array::Float64Array:
-                changed |= mergePrediction(SpecFullDouble);
+                if (node->op() == GetByVal && arrayMode.isOutOfBounds())
+                    changed |= mergePrediction(SpecBytecodeDouble | SpecOther);
```

Comment:
This is intentional. We are always boxing this into JSValue so we should not say SpecFullDouble here.

Reply:
Can we put a comment here like: "Unlike the in-bounds case, We don't need to say this is a SpecFullDouble because we're going to purify it when we box the result."